service/matching/task_queue_partition_manager.go

```diff
@@ -1303,10 +1439,25 @@ func (pm *taskQueuePartitionManagerImpl) getPerTypeUserData() (*persistencespb.T
 	return perType, userDataChanged, nil
 }
 
-func (pm *taskQueuePartitionManagerImpl) userDataChanged() {
+func (pm *taskQueuePartitionManagerImpl) userDataChanged(to *persistencespb.VersionedTaskQueueUserData) {
 	// Update rateLimits if any change is userData.
 	pm.rateLimitManager.UserDataChanged()
 
+	// Do not use defaultQueue() because that treats
+	// not being ready as an error, which is expected during bringup here.
+	defaultQ, err := pm.defaultQueueFuture.GetIfReady()
+	// Initialization error or not ready yet
+	if err != nil {
+		return
+	}
+
+	taskType := int32(pm.Partition().TaskType())
+	if to.GetData().GetPerType()[taskType].GetFairnessState() != pm.fairnessState {
+		pm.logger.Debug("unloading partitionManager due to change in FairnessState")
```

Comment:
do you want to leave this in? we already log one line for unload with cause:ConfigChange